pkg/volume/photon_pd/photon_util.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package photon_pd
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/cloudprovider"
+	"k8s.io/kubernetes/pkg/cloudprovider/providers/photon"
+	"k8s.io/kubernetes/pkg/volume"
+	volumeutil "k8s.io/kubernetes/pkg/volume/util"
+)
+
+const (
+	maxRetries         = 10
+	checkSleepDuration = time.Second
+	diskByIDPath       = "/dev/disk/by-id/"
+	diskPhotonPrefix   = "wwn-0x"
+)
+
+var ErrProbeVolume = errors.New("Error scanning attached volumes")
+var volNameToDeviceName = make(map[string]string)
```

Comment:
This is because the detacher doesn't have information about the disk ID. It only has information of volName. So we have to keep a global mapping in order to find out which device to be removed from scsi path. This will be gone when Photon start to use pvscsi controller instead of scsi controller.